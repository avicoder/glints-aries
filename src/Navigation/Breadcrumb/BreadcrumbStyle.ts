import styled from 'styled-components';
import { SecondaryColor } from '../../Utils/Colors';

export const BreadcrumbItemWrapper = styled.label<BreadcrumbItemWrapper>`
  cursor: ${({ active }) => (active ? 'default' : 'pointer')};
  font-size: 1em;
  line-height: 1.5;

  * {
    color: ${({ active }) =>
      active ? `${SecondaryColor.grey}` : `${SecondaryColor.black}`};
  }

  span {
    margin: 0 1em;
    color: ${SecondaryColor.lighterblack};
  }
`;

interface BreadcrumbItemWrapper {
  active?: boolean;
}

export const BreadcrumbContainer = styled.div`
  position: relative;
  display: flex;
  background-color: ${SecondaryColor.whitesmoke};
  padding: 1em;

  label.aries-breadcrumb-item:last-child span {
    display: none;
  }
`;
