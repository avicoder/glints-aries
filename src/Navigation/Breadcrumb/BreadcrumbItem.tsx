import * as React from 'react';
import { BreadcrumbItemWrapper } from './BreadcrumbStyle';
import classNames from 'classnames';

const BreadcrumbItem: React.FunctionComponent<Props> = (props: Props) => {
  const { active, className, children, ...defaultProps } = props;

  return (
    <BreadcrumbItemWrapper
      className={classNames('aries-breadcrumb-item', className)}
      active={active}
      {...defaultProps}
    >
      {children}
      <span>/</span>
    </BreadcrumbItemWrapper>
  );
};

interface Props
  extends React.ComponentPropsWithoutRef<typeof BreadcrumbItemWrapper> {
  children: React.ReactNode;
}

export default BreadcrumbItem;
