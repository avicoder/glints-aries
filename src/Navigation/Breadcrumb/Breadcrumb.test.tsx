import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';

import Breadcrumb from './Breadcrumb';

const oneItem = (
  <Breadcrumb>
    <Breadcrumb.Item>one</Breadcrumb.Item>
  </Breadcrumb>
);

const twoItems = (
  <Breadcrumb>
    <Breadcrumb.Item>one</Breadcrumb.Item>
    <Breadcrumb.Item>two</Breadcrumb.Item>
  </Breadcrumb>
);

describe('<Breadcrumb>', () => {
  it('should render as expected', () => {
    const oneItemSnapshot = renderer.create(oneItem).toJSON();
    expect(oneItemSnapshot).toMatchSnapshot();

    const twoItemsSnapshot = renderer.create(twoItems).toJSON();
    expect(twoItemsSnapshot).toMatchSnapshot();
  });

  it('should show correct slash count', () => {
    const { queryAllByText } = render(
      <Breadcrumb>
        <div>
          <Breadcrumb.Item>one</Breadcrumb.Item>
          <Breadcrumb.Item>two</Breadcrumb.Item>
          <Breadcrumb.Item>three</Breadcrumb.Item>
        </div>
      </Breadcrumb>
    );

    const slashes = queryAllByText('/');

    expect(slashes.length).toEqual(3);

    slashes.forEach((slash, index) => {
      if (index === slashes.length - 1) {
        expect(slash).not.toBeVisible();
      } else {
        expect(slash).toBeVisible();
      }
    });
  });
});
