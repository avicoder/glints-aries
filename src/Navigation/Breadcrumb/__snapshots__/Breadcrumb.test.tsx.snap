// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb> should render as expected 1`] = `
.c1 {
  cursor: pointer;
  font-size: 1em;
  line-height: 1.5;
}

.c1 * {
  color: #000000;
}

.c1 span {
  margin: 0 1em;
  color: #aaaaaa;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #F3F3F3;
  padding: 1em;
}

.c0 label.aries-breadcrumb-item:last-child span {
  display: none;
}

<div
  className="c0 aries-breadcrumb"
>
  <label
    className="c1 aries-breadcrumb-item"
  >
    one
    <span>
      /
    </span>
  </label>
</div>
`;

exports[`<Breadcrumb> should render as expected 2`] = `
.c1 {
  cursor: pointer;
  font-size: 1em;
  line-height: 1.5;
}

.c1 * {
  color: #000000;
}

.c1 span {
  margin: 0 1em;
  color: #aaaaaa;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #F3F3F3;
  padding: 1em;
}

.c0 label.aries-breadcrumb-item:last-child span {
  display: none;
}

<div
  className="c0 aries-breadcrumb"
>
  <label
    className="c1 aries-breadcrumb-item"
  >
    one
    <span>
      /
    </span>
  </label>
  <label
    className="c1 aries-breadcrumb-item"
  >
    two
    <span>
      /
    </span>
  </label>
</div>
`;
